### DISCUSSION TITLE
help,please

### DISCUSSION THREAD

**Discussion by tiandiren** [TPC-Council/HammerDB#537] [Q&A] [answered]
### Discussed in https://github.com/TPC-Council/HammerDB/discussions/181
when I use HammerDB 4.7 + mysql, I got this problem
<div >Error in Virtual User 1: Error: Failed to load mysqltcl - couldn't load library "C:/Program Files/HammerDB-4.7/lib/mysqltcl3.052/libmysqltcl.dll": Bad exe format. Possibly a 32/64-bit mismatch.</div>

**Comment by sm-shaw** [comment, marked as answer]
This is covered in the documentation in the section [Verifying the Installation of Database Client Libraries](https://www.hammerdb.com/docs/ch01s10.html). This also includes an example with MySQL on how to troubleshoot. 

As per the image below using the dependencies utility described in the documentation, HammerDB's libmysqltcl.dll needs to load libmysql.dll, so this needs to be in the path. Note that HammerDB is 64-bit and it shows under the Machine column as "AMD64" which is 64-bit x86. If you have a libmysql.dll in your path that is 32-bit it will give you the error you see. So make sure that the 64-bit version of the library is first in your path. 

![image](https://user-images.githubusercontent.com/38044085/232115020-f5f66ab9-d6d2-4e3a-8d91-fbb380a2e85b.png)

Use the librarycheck command to verify that the libraries can be downloaded:
```
HammerDB CLI v4.7
Copyright (C) 2003-2023 Steve Shaw
Type "help" for a list of commands
Initialized SQLite on-disk database C:/Users/Steve/AppData/Local/Temp/hammer.DB using existing tables (36,864 KB)
hammerdb>librarycheck
Checking database library for Oracle
Success ... loaded library Oratcl for Oracle
Checking database library for MSSQLServer
Success ... loaded library tdbc::odbc for MSSQLServer
Checking database library for Db2
Success ... loaded library db2tcl for Db2
Checking database library for MySQL
Success ... loaded library mysqltcl for MySQL
Checking database library for PostgreSQL
Success ... loaded library Pgtcl for PostgreSQL
Checking database library for MariaDB
Success ... loaded library mariatcl for MariaDB

hammerdb>
```
